File: app/src/main/java/io/nanodegree/andrea/popularmovies/service/PopularMoviesService.java
package io.nanodegree.andrea.popularmovies.service;

import java.io.nanodegree.popularmovies.feature.movie.data.retrofit.response.MovieContainer;

import retrofit2.Call;
import retrofit2.http.GET;

/**
 * Created by Andrea Loddo (andrea@evenly.io) on 20.12.2018
 * for Evenly GmbH
 * <p>
 * Copyright (c) 2018 Evenly GmbH,
 * all rights reserved
 */
public interface PopularMoviesService {

    @GET("popular")
    Call<MovieContainer> getListPopularMovies();

    @GET("top_rated")
    Call<MovieContainer> getListTopRatedMovies();

}

File: feature_movie/src/main/java/java/io/nanodegree/popularmovies/feature/movie/data/retrofit/PopularMoviesService.kt
package java.io.nanodegree.popularmovies.feature.movie.data.retrofit

import java.io.nanodegree.popularmovies.feature.movie.data.retrofit.response.MovieContainer

import retrofit2.Call
import retrofit2.http.GET

/**
 * Created by Andrea Loddo (andrea@evenly.io) on 20.12.2018
 * for Evenly GmbH
 *
 *
 * Copyright (c) 2018 Evenly GmbH,
 * all rights reserved
 */
interface PopularMoviesService {

    @GET("popular")
    suspend fun listPopularMovies(): MovieContainer

    @GET("top_rated")
    suspend fun listTopRatedMovies(): MovieContainer
}
